app/models/journeyDomain/transportMeans/TransportMeansDepartureDomain.scala

```diff
@@ -77,11 +90,20 @@ case class TransitionTransportMeansDepartureDomain(
 )(override val index: Index)
     extends TransportMeansDepartureDomain {
 
-  override def asString(implicit messages: Messages): String = this.toString
+  override def asString(implicit messages: Messages): String =
+    TransitionTransportMeansDepartureDomain.asString(identification, identificationNumber, index)
 }
 
 object TransitionTransportMeansDepartureDomain {
 
+  def asString(identification: Option[Identification], identificationNumber: Option[String], index: Index)(implicit messages: Messages): String =
+    (identification, identificationNumber) match {
+      case (Some(id), Some(idNumber)) => messages("departureTransportMeans.label.bothArgs", index.display, id.asString, idNumber)
+      case (Some(id), None)           => messages("departureTransportMeans.label.oneArg", index.display, id.asString)
+      case (None, Some(idNumber))     => messages("departureTransportMeans.label.oneArg", index.display, idNumber)
+      case _                          => messages("departureTransportMeans.label.noArgs", index.display)
+    }
```

Comment:
Bit of duplicated code here but no biggy. Could look to incorporate this with app/models/TransportMeans.scala but can do so in a separate PR